### ISSUE TITLE
Suggested tags should be a list

### ISSUE THREAD

**Issue by karlgroves** [WordPress/gutenberg#15291]
Suggested tags should be a list

-   **Severity:** Low
-   **Affected Populations:**
    -   Blind
    -   Low-Vision
    -   Cognitively Impaired
-   **Platform(s):**
    -   Windows - Screen Reader
    -   Mac - VoiceOver
-   **Components affected:**
    -   Publish and Unpublish

#### Issue description

Users are provided with an option to add tags when publishing, however
the tag list is not marked up as a list. Additional text elements have
been used to convey the number of items ("1 out of 3" etc.), however
these do not sufficiently compensate for the lack of structural
semantics.

Using list markup will aid comprehension for assistive technology users,
by conveying the structure of the content and providing an overview of
the number of items.

##### Issue Code

```
    <div tabindex="-1" class="...">


        <span class="components-form-token-field__token">


            <span class="components-form-token-field__token-text" id="components-form-token-field__token-text-3">


                <span class="screen-reader-text">html (1 of 3)</span>


                <span aria-hidden="true">html</span>


            </span>


            <button class="..." aria-describedby="components-form-token-field__token-text-3" aria-label="Remove Tag" type="button">


                ...


            </button>


        </span>


        <span class="components-form-token-field__token">


            <span class="components-form-token-field__token-text" id="components-form-token-field__token-text-4">


                <span class="screen-reader-text">htm (2 of 3)</span>


                <span aria-hidden="true">htm</span>


            </span>


            <button class="components-button components-icon-button components-form-token-field__remove-token" aria-describedby="components-form-token-field__token-text-4" aria-label="Remove Tag" type="button">


                ...


            </button>


        </span>


        ...


        <input class="components-form-token-field__input" id="components-form-token-input-1" role="combobox"...>


        ...


    </div>
```

#### Remediation Guidance

Convert this markup to an unordered-list (`<ul>`) with individual
list-items (`<li>`).

The `.screen-reader-text` instructions can then be removed, and the
`aria-hidden="true"` should be removed from the remaining visible
`<span>`.

##### Recommended Code

```
    <div class="components-form-token-field__input-container" tabindex="-1">


        <ul class="components-form-token-field__token">


            <li>


                <span id="components-form-token-field__token-text-0" class="...">php</span>


                <button type="button" aria-label="Remove Tag" aria-describedby="components-form-token-field__token-text-0" class="...">...</button>


            </li>


            <li>


                <span id="components-form-token-field__token-text-4" class="...">html</span>


                <button type="button" aria-label="Remove Tag" aria-describedby="components-form-token-field__token-text-4" class="...">...</button>


            </li>


            ...


        </ul>


        <input id="components-form-token-input-6" type="text" size="3" class="..." role="combobox"...>


        ...


    </div>
```

#### Relevant standards

-   [1.3.1 Info and Relationships
    (Level A)](https://www.w3.org/TR/WCAG20/#content-structure-separation-programmatic)

**Note:** This issue may be a duplicate with other existing accessibility-related bugs in this project. This issue comes from the Gutenberg accessibility audit, performed by [Tenon](https://www.tenon.io) and funded by [WP Campus](https://wpcampus.org/). This issue is GUT-14 in Tenon's report

**Comment by afercia** [comment]
Had some troubles in understanding what were the "Suggested tags" mentioned in the issue title. For clarity: is it about these ones?

<img width="282" alt="Screenshot 2019-05-02 at 19 11 20" src="https://user-images.githubusercontent.com/1682452/57093655-c7c42100-6d0e-11e9-9947-8b093bfef39f.png">

Confusion came from the fact these are actually the tags that are already _added_ to a post.

**Comment by StommePoes** [comment]
Sorry that it seems while the bugs were pasted into Github from the report, the screenshots were not. I see someone is painstakingly adding them (Mel!).

Yes it seems to match your screenshot:
![Screenshot of Add New Tag input showing two selected tags highlighted.](https://user-images.githubusercontent.com/136210/57184168-799e5180-6eb7-11e9-92d3-2a80d88dd798.png)

**Comment by afercia** [comment]
Yep. There's more to say about this component, as what appears to be an input field is actually a focusable `<div>` element visually emulating an input field. The actual input is the tiny highlighted area in the screenshot below:

<img width="300" alt="Screenshot 2019-05-05 at 23 35 19" src="https://user-images.githubusercontent.com/1682452/57200661-d6b90680-6f8e-11e9-88dd-19c3c386b902.png">

**Comment by enriquesanchez** [comment]
Noting that #21859 will implement a list view for suggested tags.

**Comment by afercia** [comment]
Hm, #21859 is about the list of _most used_ tags i.e. formerly known as the "tag cloud" in Classic Editor.

Instead, this issue is about the list of added terms. See screenshots above.

**Comment by ciampo** [comment]
@Mamaduka will https://github.com/WordPress/gutenberg/pull/80967 close this issue, too?

**Comment by Mamaduka** [comment]
It will use our new UI components under the hood, so as long as they're accessible for the Combobox + Chips pattern, it should resolve the issue.

I think it's worth double-checking.

**Comment by ciampo** [comment]
I looked more into this.

IMO, the `ul`/`li` markup is only one possible fix, and `SearchableChipSelectControl` is the right component here, although we may need to consider some tweaks.

`SearchableChipSelectControl` may need to adopt Base UI's documented multiple-select pattern: move `Combobox.Value` outside `Combobox.Chips`, then label `Combobox.Chips`, describe how to remove each `Combobox.Chip`, and give `Combobox.Input` the selected count and instructions for reaching the chips. The remove buttons are already labelled.

(also see https://github.com/mui/base-ui/pull/5628)

cc @mirka
